File: src/get_data/get_ipcs.jl
using Pkg
using CSV, DataFrames
using XLSX

wd = "/Users/christianhilscher/Desktop/tmtc/"
data_path = join([wd, "data/"])
cd(data_path)

df_ipc = DataFrame(XLSX.readdata("ipc_technology.xls", "Sheet1", "A8:H771"))

df_ipc[!, "x8"] = lowercase.(first.(df_ipc[!, "x8"], 4))

df_ipc[!, "technology"] = repeat(["complex"], size(df_ipc, 1))

for i in range(14, stop=23)
    n = size(df_ipc[df_ipc[!,"x1"].==i,:], 1)
    df_ipc[df_ipc[!,"x1"].==i,"technology"] = repeat(["discrete"], n)
end

select!(df_ipc, ["x1", "x2", "x5", "x8", "technology"])
rename!(df_ipc, [:field_num, :sector, :field, :ipc_code, :technology])

CSV.write("ipcs.csv", df_ipc)

File: src/get_data/python/get_ipcs.py
import pandas as pd 
import os 

wd_lc = "/Users/llccf/OneDrive/Dokumente/tmtc/"
os.chdir(wd_lc)

df_ipc = pd.read_excel("data/ipc_technology.xls")
df_ipc = df_ipc.iloc[6:, [0, 1, 4, 7]]
df_ipc = df_ipc.rename(columns = {"IPC8 -Technology Concordance": "field_num", 
                                "Unnamed: 4": "field", 
                                "Unnamed: 1": "sector", "Unnamed: 7": "ipc_code"}).reset_index(drop = True)

df_ipc["field_num"] = df_ipc["field_num"].astype(str).str[:4]
df_ipc["technology"] = "complex"

for i in range(14, 24): 
    field = df_ipc["field_num"] == i
    df_ipc["technology"][field] = "discrete"

df_ipc.to_csv("data/ipcs.csv")